cmd/installer/pkg/install/install.go

```diff
@@ -148,6 +148,13 @@ func Install(ctx context.Context, p runtime.Platform, mode Mode, opts *Options)
 		cmdline.SetAll(overlayOpts.KernelArgs)
 	}
 
+	// preserve console=ttyS0 if it was already present in cmdline for metal platform
+	existingCmdline := procfs.ProcCmdline()
+
+	if *existingCmdline.Get(constants.KernelParamPlatform).First() == constants.PlatformMetal && existingCmdline.Get("console").Contains("ttyS0") {
+		cmdline.Append("console", "ttyS0")
```

Comment:
if `extraKernelArgs` already has `ttyS0` this would add a duplicate, should I check there also?

Reply:
yeah, let's not do duplicates